File: infra/src/main/java/com/movie/infra/repository/TheaterJpaRepository.java
package com.movie.infra.repository;

import com.movie.domain.entity.Theater;
import com.movie.domain.repository.TheaterRepository;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface TheaterJpaRepository extends JpaRepository<Theater, Long>, TheaterRepository {
    @Override
    @Query("SELECT t.name FROM Theater t WHERE t.id = :id")
    Optional<String> findNameById(@Param("id") Long id);
}

File: data/src/main/kotlin/kr/spartacodingclub/cinema/data/repository/TheaterJpaRepository.kt
package kr.spartacodingclub.cinema.data.repository

import kr.spartacodingclub.cinema.data.entity.TheaterEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TheaterJpaRepository : JpaRepository<TheaterEntity, Long>
